### DISCUSSION TITLE
Virtualbox docker install - not enough space

### DISCUSSION THREAD

**Discussion by frankofno** [photoprism/photoprism#1540] [Q&A] [answered]
Since I could not find any more clues, how to fix the error message, "not enough space left on device" i try it here.

I've installed the app via docker-compose inside a AlmaLinux 8.4 Virtualbox container. In Virtualbox I've set the diskspace to dynamically allocated. After uploading about 2000 files the server went away. Restarting the whole Virtualbox container left me with the message, not enough space left after using docker-compose -d
With VBmanage i manually resized the disk-space (from 10 GB to 20GB). Still I'm not able to start up the docker container with compose -d 
I also found only 3000 linodes left. I guess, more will become available?

Is there a way to assign more disk space afterwards?

Thanks for any hint!

Great app, lookin forward to the face recognition feature.

**Comment by lastzero** [comment, marked as answer]
Personally haven't used Virtualbox for a long time. In general, the number of inodes depends on how you created the filesystem in the first place. Thumbnails are lots of small images, so you need more inodes than average users who edit word documents. You should still not run out of inodes unless you intentionally reduced the number or your Linux has lower defaults.

> Once the file system is created, you can not change the bytes-per-inode ratio (unless you re-format it), and resizing a filesystem changes the number of inodes to maintain this ratio. ([Source](https://www.tecmint.com/increase-disk-inode-number-in-linux/))

Note high quality thumbnails require about 30 to 40% of the storage of your originals, so 20GB for a library of 200GB won't be enough. Settings are documented here: https://docs.photoprism.org/user-guide/settings/advanced/#what-files-will-be-created-by-photoprism
